src/instana/instrumentation/pep0249.py

```diff
@@ -2,139 +2,169 @@
+
+            span.set_attribute(SpanAttributes.DB_STATEMENT, sql_sanitizer(sql))
+            span.set_attribute(SpanAttributes.DB_USER, self._connect_params[1]["user"])
+            span.set_attribute("host", self._connect_params[1]["host"])
+            span.set_attribute("port", self._connect_params[1]["port"])
         except Exception as e:
             logger.debug(e)
         return span
 
-    def __enter__(self):
+    def __enter__(self) -> CursorWrapper:
         return self
 
-    def execute(self, sql, params=None):
+    def execute(self, sql, params=None) -> None:
         tracer, parent_span, operation_name = get_tracer_tuple()
 
         # If not tracing or we're being called from sqlalchemy, just pass through
-        if (tracing_is_off() or (operation_name == "sqlalchemy")):
+        if tracing_is_off() or (operation_name == "sqlalchemy"):
             return self.__wrapped__.execute(sql, params)
 
-        with tracer.start_active_span(self._module_name, child_of=parent_span) as scope:
+        parent_context = parent_span.get_span_context() if parent_span else None
+        with tracer.start_as_current_span(
+            self._module_name, span_context=parent_context
+        ) as span:
             try:
-                self._collect_kvs(scope.span, sql)
-
+                self._collect_kvs(span, sql)
                 result = self.__wrapped__.execute(sql, params)
             except Exception as e:
-                if scope.span:
-                    scope.span.log_exception(e)
+                if span:
+                    span.record_exception(e)
                 raise
             else:
                 return result
 
-    def executemany(self, sql, seq_of_parameters):
+    def executemany(self, sql, seq_of_parameters) -> None:
         tracer, parent_span, operation_name = get_tracer_tuple()
 
         # If not tracing or we're being called from sqlalchemy, just pass through
-        if (tracing_is_off() or (operation_name == "sqlalchemy")):
+        if tracing_is_off() or (operation_name == "sqlalchemy"):
             return self.__wrapped__.executemany(sql, seq_of_parameters)
 
-        with tracer.start_active_span(self._module_name, child_of=parent_span) as scope:
+        parent_context = parent_span.get_span_context() if parent_span else None
+        with tracer.start_as_current_span(
+            self._module_name, span_context=parent_context
+        ) as span:
             try:
-                self._collect_kvs(scope.span, sql)
-
+                self._collect_kvs(span, sql)
                 result = self.__wrapped__.executemany(sql, seq_of_parameters)
             except Exception as e:
-                if scope.span:
-                    scope.span.log_exception(e)
+                if span:
+                    span.record_exception(e)
                 raise
             else:
                 return result
 
-    def callproc(self, proc_name, params):
+    def callproc(self, proc_name, params) -> None:
```

Comment:
Missing type hints for the arguments here, and the return must be corrected.

Reply:
callproc returns None 

![image](https://github.com/user-attachments/assets/7e06e968-b588-4747-a172-c91938c3b5af)